marklogic-data-hub-central/ui/src/components/entities/mapping/source-entity-map/source-to-entity-map.test.tsx

```diff
@@ -259,6 +266,120 @@ describe('RTL Source-to-entity map tests', () => {
+    expect(firstName[1]).toHaveStyle("line-height: 2vh; text-indent: 20px;");
+
+    //Click on 'FirstNamePreferred'
+    fireEvent.click(getAllByText("FirstNamePreferred")[1]);
+
+    //mapping is saved
+    expect(await(waitForElement(() => getAllByTestId("successMessage").length,{"timeout":200}))).toEqual(1)
+
+    let mapExp = getAllByTestId("mapexpression");
+    //Right Xpath is populated
+    expect(mapExp[3]).toHaveTextContent("nutFreeName/FirstNamePreferred");
+
+});
+
+test('Nested source data/entity - Right XPATH with source context',  async() => {
+    axiosMock.post.mockImplementationOnce(data.mapProps.updateMappingArtifact);
+    const {getAllByText,getAllByTestId, getAllByRole,debug } = render(<SourceToEntityMap {...data.mapProps}  mappingVisible={true}/>);
+
+    //Ensure source selectors are present
+    let sourceSelectors = getAllByTestId("listSources");
+    expect(sourceSelectors.length).toEqual(8)
+
+    // Set 'sourceContext' for ItemType entity
+    fireEvent.click(sourceSelectors[2]);
+    await(waitForElement(() =>  getAllByRole("option"),{"timeout":200}))
+    //Set 'sourceContext' to 'nutFreeName'
+    fireEvent.click(getAllByText("nutFreeName")[1]);
+    expect(await(waitForElement(() => getAllByTestId("successMessage").length,{"timeout":200}))).toEqual(1)
+    let mapExp = getAllByTestId("mapexpression");
+    //Right Xpath is populated
+    expect(mapExp[2]).toHaveTextContent("nutFreeName");
+
+
+    //Click on the fourth selector (FirstNamePreferred)
+    fireEvent.click(sourceSelectors[3]);
+    await(waitForElement(() => getAllByRole("option"),{"timeout":400}))
+    let firstName = getAllByText("FirstNamePreferred");
+    fireEvent.click(firstName[1]);
+    //mapping is saved
+    expect(await(waitForElement(() => getAllByTestId("successMessage").length,{"timeout":200}))).toEqual(1)
+    mapExp = getAllByTestId("mapexpression");
+
+    //Right Xpath is populated (and not nutFreeName/FirstNamePreferred since sourceContext is set)
+    expect(mapExp[3]).toHaveTextContent("FirstNamePreferred");
+
+});
+
+test('Search source',  async() => {
+    axiosMock.post.mockImplementationOnce(data.mapProps.updateMappingArtifact);
+    const {getByText,getAllByText, getAllByTestId, getAllByRole} = render(<SourceToEntityMap {...data.mapProps}  mappingVisible={true}/>);
+
+    //Ensure source selectors are present
+    let sourceSelectors = getAllByTestId("listSources");
+
+    expect(sourceSelectors.length).toEqual(8)
+
+    //corresponds to 'id' source selector
+    fireEvent.click(sourceSelectors[3]);
+
+    await(waitForElement(() =>  getAllByRole("option"),{"timeout":200}))
+    let firstName = getAllByText("FirstNamePreferred");
+    expect(firstName.length).toEqual(2)
+
+    let lastName = getAllByText("LastName");
+    expect(lastName.length).toEqual(2)
+
+    let inputBox = getByText(
+        (_content, element) =>
+            element.className != null &&
+            element.className ==="ant-select-search__field"
```

Comment:
fx search also has the same className. Can we ensure we are getting the correct inputBox here

Reply:
The point is only one of them should be rendered in the UI at any time and the one rendered should be the one I click. So, I believe, this is ok